### PR TITLE
106 admin login

### PR DESCRIPTION
**Pull request by abdou059**

### CODE REVIEW

File: client/src/pages/AdminLogin/index.tsx [hunk 1]
```diff
@@ -1,5 +1,93 @@
...
+  const { setUserInfo }:UserContextTypeWithDispatch = useContext(UserContext);
+  const navigate = useNavigate();
+  const formik = useFormik({
+    initialValues: {
+      UserName: '',
+      password: '',
+    },
+    validationSchema: loginSchema,
+    onSubmit: (values) => {
+      const login = async () => {
```
Comment (husamkamal) [inline client/src/pages/AdminLogin/index.tsx]: remove the auth/login to admin endpoint

File: client/src/helpers/validationSchema.ts [hunk 2]
```diff
@@ -29,4 +29,9 @@ const loginSchema = yup.object({
     .min(8, 'Password should be of minimum 8 characters length')
     .required('Password is required'),
 });
-export { loginSchema, addCarSchema };
+const AdminLoginSchema = yup.object({
```
Comment (LinaYahya) [inline client/src/helpers/validationSchema.ts]: What about password?

File: client/src/helpers/validationSchema.ts [hunk 3]
```diff
@@ -29,4 +29,9 @@ const loginSchema = yup.object({
...
+  userName: yup
```
Comment (LinaYahya) [inline client/src/helpers/validationSchema.ts]:
```suggestion
  username: yup
```

File: client/src/pages/AdminLogin/index.tsx [hunk 4]
```diff
@@ -1,5 +1,83 @@
...
+import { UserContextTypeWithDispatch } from '../../interfaces';
+
 function AdminLogin() {
-  return (<h1>AdminLogin</h1>);
+  const [responseError, setResponseError] = useState<string>('');
+  const { setUserInfo }:UserContextTypeWithDispatch = useContext(UserContext);
+  const navigate = useNavigate();
+  const formik = useFormik({
+    initialValues: {
+      UserName: '',
```
Comment (LinaYahya) [inline client/src/pages/AdminLogin/index.tsx]:
```suggestion
      Username: '',
```

File: client/src/pages/AdminLogin/index.tsx [hunk 5]
```diff
@@ -1,5 +1,83 @@
...
+            buy or sell a used car
+          </h2>
+          <p>Great Value | Trusted Quality | All Online</p>
+        </div>
+      </div>
+      <div className="formPage">
+        <form className="formLogin" onSubmit={formik.handleSubmit}>
+          <h1>LOG IN</h1>
+          <TextField
+            id="outlined-basic"
```
Comment (LinaYahya) [inline client/src/pages/AdminLogin/index.tsx]: Why there's no value for username

**Review by LinaYahya — CHANGES_REQUESTED** [review]

File: client/src/pages/AdminLogin/index.tsx (hunk 5 above)
Comment (LinaYahya) [reply client/src/pages/AdminLogin/index.tsx]: and onChange and error, helper text?

**Review by LinaYahya — CHANGES_REQUESTED** [review]

File: client/src/pages/AdminLogin/index.tsx [hunk 6]
```diff
@@ -1,5 +1,99 @@
...
+  const { setUserInfo }:UserContextTypeWithDispatch = useContext(UserContext);
+  const navigate = useNavigate();
+  const formik = useFormik({
+    initialValues: {
+      username: '',
+      password: '',
+    },
+    validationSchema: AdminLoginSchema,
+    onSubmit: (values) => {
+      const Admin = async () => {
```
Comment (LinaYahya) [inline client/src/pages/AdminLogin/index.tsx]:
```suggestion
      const Admin = async () => {
```
Not sure what function name means?

File: client/src/pages/AdminLogin/index.tsx [hunk 7]
```diff
@@ -1,5 +1,99 @@
...
+        try {
+          const result = await httpInstance.post('/admin/login', values);
+          setUserInfo(result.data);
+          console.log(result);
```
Comment (LinaYahya) [inline client/src/pages/AdminLogin/index.tsx]:
```suggestion
```

File: client/src/pages/AdminLogin/index.tsx [hunk 8]
```diff
@@ -1,5 +1,99 @@
...
+
+          navigate('/');
+        } catch (error) {
+          setResponseError('error');
```
Comment (LinaYahya) [inline client/src/pages/AdminLogin/index.tsx]:
```suggestion
          setResponseError('error'); // IS that a clear and understood error message?
```

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by mai-mohammed — APPROVED** [review]

File: client/src/pages/AdminLogin/index.tsx [hunk 9]
```diff
@@ -1,5 +1,97 @@
...
+import { useNavigate } from 'react-router-dom';
+import { UserContext } from '../../context';
+import { AdminLoginSchema } from '../../helpers/validationSchema';
+import { UserContextTypeWithDispatch } from '../../interfaces';
+import httpInstance from '../../services';
+import './styles.css';
+
 function AdminLogin() {
-  return (<h1>AdminLogin</h1>);
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (tariqabusada) [inline client/src/pages/AdminLogin/index.tsx]: no need to unused variables so we don't need the 'disable' comment

File: client/src/pages/AdminLogin/index.tsx [hunk 10]
```diff
@@ -1,5 +1,97 @@
...
+  const { setUserInfo }:UserContextTypeWithDispatch = useContext(UserContext);
+  const navigate = useNavigate();
+  const formik = useFormik({
+    initialValues: {
+      username: '',
+      password: '',
+    },
+    validationSchema: AdminLoginSchema,
+    onSubmit: (values) => {
+      const AdminInfo = async () => {
```
Comment (LinaYahya) [inline client/src/pages/AdminLogin/index.tsx]: not sure why you create a function inside onSubmit, either to create a function outside the component or to just call request outside of function

**Review by husamkamal — APPROVED** [review]

File: client/src/pages/AdminLogin/index.tsx [hunk 11]
```diff
@@ -1,5 +1,96 @@
...
+            name="password"
+            label="Password"
+            type="password"
+            value={formik.values.password}
+            onChange={formik.handleChange}
+            error={formik.touched.password && Boolean(formik.errors.password)}
+            helperText={formik.touched.password && formik.errors.password}
+          />
+          {
+            responseError
```
Comment (LinaYahya) [inline client/src/pages/AdminLogin/index.tsx:74]: try to set response error after reset or update form values

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by LinaYahya — APPROVED** [review]

**Review by tariqabusada — APPROVED** [review]

**Pull request merged**